app/controllers/api_controller.rb

```diff
@@ -91,32 +103,97 @@ def render_error(resource, status)
            serializer: ActiveModel::Serializer::ErrorSerializer
   end
 
-  def get_user_from_request(request)
-    # TODO: this block can be removed once CT is fully removed
-    if request.params[:loggedUser].present?
-      if request.params[:loggedUser].is_a? String
-        return JSON.parse(request.params[:loggedUser]) || {}
-      else
-        return request.params[:loggedUser] || {}
-      end
+  def validate_request(request)
+    logger.debug 'Validating request'
+
+    if request.headers['Authorization'].nil?
+      logger.debug 'No authorization header found'
+    end
+
+    if request.headers['x-api-key'].nil? && ENV.fetch('REQUIRE_API_KEY', true)
+      raise ApiKeyError.new()
+    end
+
+    body = {}
+
+    if request.headers['Authorization']
+      body['userToken'] = request.headers['Authorization']
+    end
+
+    if request.headers['x-api-key']
+      body['apiKey'] = request.headers['x-api-key']
     end
 
-    if request.headers["authorization"].nil?
-      return {}
+    if body.present?
+      headers = {
+        'Authorization' => "Bearer #{ENV.fetch('MICROSERVICE_TOKEN')}"
+      }
+
+      response = HTTParty.post("#{ENV.fetch('GATEWAY_URL')}/v1/request/validate", body: body, headers: headers)
```

Comment:
this bit, wondering if you might need some exception handling, e.g. when no connectivity?